### PR TITLE
Surface temperature model

### LINKED ISSUE #4022: Add "Surface temperature" submodel
This model accounts for the fact that the surface temperature of a cell in a thermal chamber is typically different from the "set"/"ambient" temperature of the chamber.

Suggested steps to do this below, but open to discussion:
- Add a submodel `SurfaceAmbientTemperature` which sets surface temperature equal to ambient temperature
- Add another submodel `SurfaceTemperatureODE` which implements the equation from [Mohan et al](https://d1wqtxts1xzle7.cloudfront.net/69868813/j.jpowsour.2014.01.09720210918-12240-190ns8h-libre.pdf?1632013230=&response-content-disposition=inline%3B+filename%3DA_lumped_parameter_electro_thermal_model.pdf&Expires=1713470219&Signature=VVdyggn0EMZaZUDAhblutOENZ6faE0Wua0uoq71ebx912paMfn2O2-lJq8SUK1nwkZf7v5jF65dIUF03R~IKZo5LXZdYEOb48oPshUycdG0h3K4EHPaGW07CNjd6e~UFP2JyNjUDMAnfP5oZHD~8cr5MRoTbNi8WTfcJ8SSdc1nyOr4uKy5xevjQKW3D0cvAbu9gSnbVZC52qdwM6jeobcJLAx7AhMetR3x1r2lHd6Jrrz-vznm1JeczXl6EzEzMgqr2tLOcKO7GLhlUVYIDhdGawR76FJHLBNyhAEPXtEMq7IUibswa~~2fdgZNTemj9lCjsExrmrri-xZhHFL5vQ__&Key-Pair-Id=APKAJLOHF5GGSLRBV4ZA):
Screenshot 2024-04-18 at 2 59 00 PM

- In that submodel, register a citation for that paper
- In all other thermal submodels, replace "ambient temperature" with "surface temperature"
- In `BaseBatteryModel`, add an option for "surface temperature" which can be "ambient" (default) or "ODE", then add an if statement for selecting the appropriate model
- Add a test for selecting the "ODE" option
- Add docs

### PR DESCRIPTION
**Pull request by valentinsulzer**
# Description

From Lin et al 2014 (kudos @js1tr3 et al) 
image
image

Fixes #4022 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by valentinsulzer** [comment]
Still need to add tests and update examples

**Comment by TomTranter** [comment]
I am happy to review this when ready

**Comment by valentinsulzer** [comment]
@TomTranter ready for review

**Comment by kratman** [comment]
Looks like the examples are failing here:
```
            for phase in self.options.phases[domain]:
                if particle == "Fickian diffusion":
                    submod = pybamm.particle.FickianDiffusion(
                        self.param, domain, self.options, phase=phase, x_average=True
                    )
                elif particle in [
                    "uniform profile",
                    "quadratic profile",
                    "quartic profile",
                ]:
                    submod = pybamm.particle.XAveragedPolynomialProfile(
                        self.param, domain, self.options, phase=phase
                    )
                elif particle == "MSMR":
                    submod = pybamm.particle.MSMRDiffusion(
                        self.param, domain, self.options, phase=phase, x_average=True
                    )
                    # also set the submodel for calculating stoichiometry from
                    # potential
                    self.submodels[f"{domain} {phase} stoichiometry"] = (
                        pybamm.particle.MSMRStoichiometryVariables(
                            self.param,
                            domain,
                            self.options,
                            phase=phase,
                            x_average=True,
                        )
                    )
                self.submodels[f"{domain} {phase} particle"] = submod
```

The variable submod does not get defined before assignment. PyCharm identifies that the variable may be undefined. We should probably make sure our inspection tools flag this

**Comment by TomTranter** [comment]
The model seems to be working but I am not sure about some of the numbers or what the constants in the equation represent. It would make more sense to me to have 3 nodes representing the core, casing and environment. I also would prefer the equations to be consistent with the other heat capacities and specify the thermal conductivity and specific heat capacity. The paper quotes these values for the parameters in the model 
image
The thermal resistance seems ok but capacity seems a bit high unless my assumed casing thickness is a bit low.
```
# Assuming a 21700 cylindrical cell with Al casing

rho_Al = 2710.00  # [kg/m^3]
h_casing = 70.0e-3  # [m] height of casing
t_casing = 1.0e-3  # [m] casing thickness
r_casing = 10.50e-3  # [m] casing radius

# Volume of the cylindrical side
vol_casing_side = np.pi * ((r_casing + t_casing)**2 - r_casing**2) * h_casing  # [m^3]

# Volume of the end caps
vol_casing_caps = 2 * np.pi * r_casing**2 * t_casing  # [m^3]

# Total volume of the casing
vol_casing = vol_casing_side + vol_casing_caps

Cp_Al = 890.0  # [J/kg.K]
C_Al = rho_Al * vol_casing * Cp_Al

print("Casing heat capacity [J/K]:", C_Al)


k_Al = 237 # W.m-1K-1
R_casing = 1 / (k_Al * t_casing)

print("Environment thermal resistance [K/W]:", R_casing)

parameter_values = pybamm.ParameterValues("Chen2020")
parameter_values.update(
    {
        "Casing heat capacity [J.K-1]": C_Al,
        "Environment thermal resistance [K.W-1]": R_casing,
    },
    check_already_exists=False,
)
```
Gives
Casing heat capacity [J/K]: 13.33
Environment thermal resistance [K/W]: 4.21

Could we see what adding another equation does to the model to go from casing to cooling channel centre through air. It seems like the fitted parameters for this model might be doing some averaging of properties which is not that obvious to the user

**Comment by valentinsulzer** [comment]
I'd like to stick as closely as possible to what is implemented in the referred paper, so that we can say "this is the model from this paper" rather than having to justify choices of equations and parameters. Especially as this model already gives good fit to some data I've been looking at

File: pybamm/models/full_battery_models/base_battery_model.py [hunk 1]
```diff
@@ -194,6 +194,11 @@ class BatteryModelOptions(pybamm.FuzzyDict):
             * "surface form" : str
                 Whether to use the surface formulation of the problem. Can be "false"
                 (default), "differential" or "algebraic".
+            * "surface temperature" : str
+                Sets the surface temperature model to use. Can be "ambient" (default),
+                which sets the surface temperature equal to the ambient temperature, or
+                "lumped", which adds an ODE for the surface temperature (e.g. to model)
```
Comment (rtimms) [inline pybamm/models/full_battery_models/base_battery_model.py:200]:
```suggestion
                "lumped", which adds an ODE for the surface temperature (e.g. to model
```

File: pybamm/models/submodels/thermal/surface/lumped.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#
+# Class for ambient surface temperature submodel
+import pybamm
+
+
+class Lumped(pybamm.BaseSubModel):
+    """
+    Class for the lumped surface temperature submodel, which adds an ODE for the
+    surface temperature.
```
Comment (rtimms) [inline pybamm/models/submodels/thermal/surface/lumped.py:9]: might be worth putting the citation in the docs

**Review by rtimms — APPROVED** [review]
thanks @valentinsulzer ! looks good to me once the conflicts are fixed

File: pybamm/models/full_battery_models/base_battery_model.py (hunk 1 above)
Comment (kratman) [reply pybamm/models/full_battery_models/base_battery_model.py]: Done

**Pull request merged**